ocsp.go

```diff
@@ -634,7 +616,8 @@ func initOCSPCache() {
 		}
 	}
 	for k, cacheValue := range buf {
-		if extractOCSPCacheResponseValue(cacheValue) == ocsp.Unknown {
+		status := extractOCSPCacheResponseValueWithoutSubject(cacheValue)
```

Comment:
Right now we are validating OCSP Responses multiple times. During init we parse and validate all OCSP Responses completely. This is slightly similar to Python where we validate the timestamp and the time validity of the OCSP Responses. Trying to remember why we decided to do this.

Reply:
At init time, the OCSP response cache is read from a local disk, which may have invalid OCSP response. When downloading the OCSP response cache, the contents may be invalid, so do we.

Reply:
But this should be taken care when the driver actually needs that OCSP Response right? Otherwise the driver doesn't really care if the local cache has bad responses. The only case it would care would be in Python driver and that too when the driver is being used by the OCSP Cache Server because we don't want the caches to have bad responses.

Reply:
yeah, this is a kind of some optimization. For cost and code complexity wise, it is not big deal here. Both validations and the actual validations code share the same code path in Go driver.
Python Connector has duplications, so we can simplify it by refactoring codes.